src/core_plugins/kibana/public/home/components/sample_data_set_card.js

```diff
@@ -58,36 +67,53 @@ export class SampleDataSetCard extends React.Component {
       isProcessingRequest: true,
     });
 
-    if (this.isInstalled()) {
-      await uninstallSampleDataSet(id, name, defaultIndex, getConfig, setConfig, clearIndexPatternsCache);
-    } else {
-      await installSampleDataSet(id, name, defaultIndex, getConfig, setConfig, clearIndexPatternsCache);
-    }
+    const isSuccess = await installSampleDataSet(id, name, defaultIndex, getConfig, setConfig, clearIndexPatternsCache);
+
+    this.setState({
+      isProcessingRequest: false,
+      status: isSuccess ? INSTALLED_STATUS : UNINSTALLED_STATUS
+    });
+  }
+
+  uninstall = async () => {
+    const {
+      getConfig,
+      setConfig,
+      id,
+      name,
+      defaultIndex,
+      clearIndexPatternsCache,
+    } = this.props;
+
+    this.setState({
+      isProcessingRequest: true,
```

Comment:
This is just another status.  Can we merge the props into one?  Status = PROCESSING, INSTALLED, UNINSTALLED.  and I guess maybe UNKOWN?

How do I simulate UNKNOWN?  I can't seem to trigger it.

I think even if you handle the isMounted checks below you can get into a weird state by hitting 'Add' or 'Remove' the quickly switching tabs and going back.  isProcessingREquest is state on this button so that gets lost but the request is still processing.  Then I think you will end up with indicating the wrong status.

<img width="1248" alt="screen shot 2018-08-06 at 2 01 50 pm" src="https://user-images.githubusercontent.com/16563603/43733032-51ee06d2-9981-11e8-9554-d583bea118bf.png">

Reply:
`UNKNOWN ` surfaces if there are any problems accessing the data, like permission problems